src/jekyll/jk_document.rs

```diff
@@ -0,0 +1,290 @@
+        Ok(JkDocument {
+               front: front.map(|s| s.to_owned()),
+               content: content.to_owned(),
+           })
+    }
+
+    pub fn parse(source_file: &Path) -> Result<JkDocument> {
+        let doc: String = read_file(source_file)?;
+        JkDocument::parse_string(doc)
+    }
+
+    pub fn convert_front(front: String) -> Result<String> {
+        let front_value: JkFrontmatterBuilder = serde_yaml::from_str(&front)?;
+        let front_builder: FrontmatterBuilder = front_value.into();
+        let front = front_builder.build()?;
+        let mut converted = serde_yaml::to_string(&front)?;
+        converted.drain(..4);
+        Ok(converted)
+    }
+
+    pub fn convert(source_file: &Path, dest_dir: &Path) -> Result<()> {
+        let doc = JkDocument::parse(source_file)?;
+        let front = match doc.front {
+            Some(front) => Some(JkDocument::convert_front(front)?),
+            None => None,
+        };
+
+        if !dest_dir.exists() {
+            create_dir_all(&dest_dir)?;
+        }
+        let dest_file = dest_dir.join(source_file.with_extension("md").file_name().unwrap());
+        let mut dest = File::create(dest_file)?;
+        let converted = match front {
+            Some(front) => format!("{}\n---\n{}", &front, &doc.content),
+            None => doc.content,
+        };
+        dest.write_all(converted.as_bytes())?;
+        Ok(())
+    }
+}
+
+pub fn convert_from_jk(source: &Path, dest: &Path) -> Result<()> {
+    if dest.is_file() {
+        Err(ErrorKind::CantOutputInFile.into())
+    } else if source.is_file() {
+        JkDocument::convert(source, dest)
+    } else if source.is_dir() {
+        for file in source.read_dir()? {
+            if let Ok(file) = file {
+                let file_path = file.path();
+                let ext = file_path.extension().unwrap_or(OsStr::new(""));
+                if file_path.is_file() {
+                    if ext == "md" || ext == "markdown" {
+                        JkDocument::convert(&file.path(), dest)?
+                    } else {
+                        warn!("unsupported file extension")
+                    }
+                } else {
+                    warn!("sub directory parsing is not supported yet")
+                }
+            }
+        }
+        Ok(())
+    } else {
+        Err(ErrorKind::InternalError.into())
+    }
+}
+
+fn read_file<P: AsRef<Path>>(path: P) -> Result<String> {
+    let mut file = File::open(path.as_ref())?;
+    let mut text = String::new();
+    file.read_to_string(&mut text)?;
+    Ok(text)
+}
+
+fn split_document(content: &str) -> Result<(Option<&str>, &str)> {
+    if FRONT_MATTER_DIVIDE.is_match(content) {
+        let mut splits = FRONT_MATTER_DIVIDE.splitn(content, 3);
+        let first = splits.next().unwrap_or("");
+        let second = splits.next().unwrap_or("");
+        let third = splits.next().unwrap_or("");
+
+        if third.is_empty() {
+            // only one "---"
```

Comment:
Is there a reason you ended up supporting this use case?  Does jekyll actually support it?

Reply:
I don't know ^^' I'm fairly new to the world of static web site generator. I wanted to be robust here. I'll try to find some info and maybe find out that an error is more appropriate :)

Reply:
Thanks.

Reply:
If I believe Jekyll's documentation on frontmatter https://jekyllrb.com/docs/frontmatter/

> The front matter is where Jekyll starts to get really cool. Any file that contains a YAML front matter block will be processed by Jekyll as a special file. The front matter must be the first thing in the file and must take the form of valid YAML set between triple-dashed lines.

So it seems that I don't need to manage this but spit an error instead :)

Reply:
Already made the change in #298 as well as fix some bugs I forgot about during the review.